File: core/src/main/java/ru/surfstudio/android/core/app/interactor/common/network/ServerUrlModule.java
package ru.surfstudio.android.core.app.interactor.common.network;


import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.core.app.dagger.scope.PerApplication;

/**
 * Модуль предоставления url сервера
 */
@Module
public class ServerUrlModule {

    @Provides
    @PerApplication
    String provideBaseApiUrl() {
        //todo

//        return BuildConfig.USE_PRODUCTION_URL
//                ? ServerUrls.PRODUCTION_API_URL
//                : ServerUrls.TEST_API_URL;
        return "";
    }
}

File: shared-pref/sample/src/main/java/ru/surfstudio/android/shared/pref/sample/interactor/common/network/ServerUrlModule.kt
package ru.surfstudio.android.shared.pref.sample.interactor.common.network

import dagger.Module
import dagger.Provides
import ru.surfstudio.android.dagger.scope.PerApplication
import ru.surfstudio.android.network.BaseUrl

@Module
class ServerUrlModule {

    companion object {
        private const val BASE_URL = "https://api.ipify.org"
    }

    @Provides
    @PerApplication
    fun provideBaseUrl(): BaseUrl = BaseUrl(BASE_URL, null)
}